File: app/src/main/java/vn/com/frankle/karaokelover/KAppComponent.java
package vn.com.frankle.karaokelover;

import android.support.annotation.NonNull;

import com.pushtorefresh.storio.sqlite.StorIOSQLite;

import javax.inject.Singleton;

import dagger.Component;
import vn.com.frankle.karaokelover.database.DbModule;
import vn.com.frankle.karaokelover.fragments.KFragmentHome;

@Singleton
@Component(
        modules = {
                KAppModule.class,
                DbModule.class
        }
)
public interface KAppComponent {
    void inject(@NonNull KFragmentHome fragmentHome);

    @NonNull
    StorIOSQLite storIOSQLite();
}

File: app/src/main/java/vn/com/frankle/karaokelover/di/KAppComponent.kt
package vn.com.frankle.karaokelover.di

import com.droidcba.kedditbysteps.di.AppModule
import com.pushtorefresh.storio.sqlite.StorIOSQLite
import dagger.Component
import vn.com.frankle.karaokelover.KActivitySearch
import vn.com.frankle.karaokelover.fragments.KFragmentFavorite
import vn.com.frankle.karaokelover.fragments.KFragmentHome
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(
        AppModule::class,
        DatabaseModule::class)
)
interface KAppComponent {
    fun inject(fragmentHome: KFragmentHome)
    fun inject(fragmentFavorite: KFragmentFavorite)
    fun inject(activitySearch: KActivitySearch)

    fun storIOSQLite(): StorIOSQLite
}
